packages/fcl/src/wallet-provider-spec/draft-v3.md

```diff
@@ -0,0 +1,548 @@
+```
+
+During authentication, the application has a chance to request to you, the wallet, what they would like you to send back to them. These requests are included in the `FCL:FRAME:READY:RESPONSE` messsage sent to the wallet from FCL.
+
+An example of such a request is the OpenID service. The application can request for you to send them the email address of the current user. The application requesting this information does not mean you need to send it. It's entirely optional for you to do so.
+
+In the config they can also tell you a variety of things about them, such as the name of their application or a url for an icon of their application. You can use these pieces of information to customize your wallets user experience should you desire to do so.
+
+Your wallet having a visual distinction from the application, but still a seemless and connected experience is our goal here.
+
+Whether your authentication process happens using a webpage using the `IFRAME/RPC`, `POP/RPC` or `TAB/RPC` methods, or using a backchannel to an API with the `HTTP/POST` method, the handshake is the same. The same messages are sent in both methods, however the transport mechanism changes. For `IFRAME/RPC`, `POP/RPC` or `TAB/RPC` methods, the transport is window post messsages, with the `HTTP/POST` method, the tranport is HTTP post messages. 
+
+As always, you must never trust anything you recieve from an application. Always do your due-dilligence and be alert as you, the wallet, are the users first line of defense against potentially malicious applications.
+
+### Authenticate your User 
+
+It's important that you are confident that the user is who the user claims to be.
+
+Have them provide enough proof to you that you are okay with passing their details back to FCL.
+If we were to use Blocto as an example, this is when the user is logging into Blocto, putting in their email and getting sent an authentication code to their email. Everything Blocto needs to do to be confident in the users identity. 
+
+### Once you know who your User is
+
+Once you're confident in the users identity, we can complete the authentication process.
+
+The authentication process is complete once FCL receives back a response, via a post message, that configures FCL with services (more on this concept later) for the current user. This response is extremeley important to FCL. At its core it tells FCL who the user is, and then via included services it tells FCL how the user authenticated, how to request transaction signatures, how to get a personal message signed and the user's email and other details if requested. In the future it may also inlude many more things!
+
+You can kind of think of FCL as a plugin system. But since those plugins exist elsewhere outside of FCL, FCL needs to be configured with information on how to communicate with those plugins.
+
+What you are sending back to FCL is everything that it needs to communicate with the plugins that you, the wallet, are supplying.
```

Comment:
"that you, the wallet," Still stands out to me 
Maybe we define a few terms at the intro (like a Contract) so there's no need to repeat.

Reply:
Good idea! Ill add blurb about who "you" is referring to in the doc at the top